apps/dashboard/src/components/workflow-editor/step-default-values.ts

```diff
@@ -1,14 +1,22 @@
-import { StepResponseDto } from '@novu/shared';
 import { buildDefaultValues, buildDefaultValuesOfDataSchema } from '@/utils/schema';
+import { StepResponseDto } from '@novu/shared';
 
 // Use the UI Schema to build the default values if it exists else use the data schema (code-first approach) values
 export const getStepDefaultValues = (step: StepResponseDto): Record<string, unknown> => {
   const controlValues = step.controls.values;
-  const hasControlValues = Object.keys(controlValues).length > 0;
+
+  const uiSchemaDefaultValues = buildDefaultValues(step.controls.uiSchema ?? {});
+  const dataSchemaDefaultValues = buildDefaultValuesOfDataSchema(step.controls.dataSchema ?? {});
 
   if (Object.keys(step.controls.uiSchema ?? {}).length !== 0) {
-    return hasControlValues ? controlValues : buildDefaultValues(step.controls.uiSchema ?? {});
+    return {
+      ...uiSchemaDefaultValues,
+      ...controlValues,
+    };
```

Comment:
why we needed to update this?

Reply:
There's a bug where if you visit step conditions and make a change before touching the template, your app crashes when visiting the template afterwards.